### PR TITLE
Splunk app 2.0.4, support url in the setup, added url regex

### PR DESCRIPTION
**Pull request by hirshbergshachar**

### CODE REVIEW

File: add-on/TA-Demisto/bin/demisto_setup.py [hunk 1]
```diff
@@ -21,7 +22,8 @@
 CONFIG_ENDPOINT = "/servicesNS/nobody/TA-Demisto/configs/conf-demistosetup/demistoenv/"
 PORT_REGEX = "^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$"
 IP_REGEX = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$"
-DOMAIN_REGEX = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])*\/{0,1}[A-Za-z0-9][A-Za-z0-9\-]*\_{0,1}[A-Za-z0-9][A-Za-z0-9\-]*$"
+DOMAIN_REGEX = "(?i)(?:[-A-Z0-9]+\[?\.\]?)+[-A-Z0-9]+(?::[0-9]+)?(?:(?:\/|\?)[-A-Z0-9+&@#\/%=~_$?!\-:,.\(\);]*[A-Z0-9+&@#\/%=~_$\(\);])?|\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b"
+URL_REGEX = "(?i)(?:(?:https?|ftp|hxxps?):\/\/|www\[?\.\]?|ftp\[?\.\]?)(?:[-A-Z0-9]+\[?\.\]?)+[-A-Z0-9]+(?::[0-9]+)?(?:(?:\/|\?)[-A-Z0-9+&@#\/%=~_$?!\-:,.\(\);\*–—―|]*[A-Z0-9+&@#\/%=~_$\(\);\*–—―|])?"
```
Comment (orius123) [inline add-on/TA-Demisto/bin/demisto_setup.py]: Where are these regexps from? We don't need ftp/hxxp here

File: add-on/TA-Demisto/bin/demisto_setup.py [hunk 2]
```diff
@@ -21,7 +22,8 @@
 CONFIG_ENDPOINT = "/servicesNS/nobody/TA-Demisto/configs/conf-demistosetup/demistoenv/"
 PORT_REGEX = "^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$"
 IP_REGEX = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$"
-DOMAIN_REGEX = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])*\/{0,1}[A-Za-z0-9][A-Za-z0-9\-]*\_{0,1}[A-Za-z0-9][A-Za-z0-9\-]*$"
+DOMAIN_REGEX = "(?i)(?:[-A-Z0-9]+\[?\.\]?)+[-A-Z0-9]+(?::[0-9]+)?(?:(?:\/|\?)[-A-Z0-9+&@#\/%=~_$?!\-:,.\(\);]*[A-Z0-9+&@#\/%=~_$\(\);])?|\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b"
```
Comment (orius123) [inline add-on/TA-Demisto/bin/demisto_setup.py:25]: We should not support domain without address

File: add-on/TA-Demisto/bin/demisto_setup.py [hunk 3]
```diff
@@ -422,15 +424,17 @@ def create_servers_config(self, urls, ports, passwords, certificates):
                 splitted_urls) + " Current passwords: " + str(splitted_passwords))
 
         for url in splitted_urls:
-            server_url = 'https://'
```
Comment (orius123) [inline add-on/TA-Demisto/bin/demisto_setup.py:425]: We still need https verification for IPs

Comment (orius123) [reply add-on/TA-Demisto/bin/demisto_setup.py:425]: Ohh. Saw you add it afterwards

File: add-on/TA-Demisto/default/app.conf [hunk 4]
```diff
@@ -11,7 +11,7 @@ label = Demisto Add-on for Splunk
 [launcher]
 author = Demisto Inc
 description = This application provides an alert action to create an incident in Demisto.
-version = 2.0.3
+version = 2.0.4
```
Comment (orius123) [inline add-on/TA-Demisto/default/app.conf:15]: Do we build and CI already?

File: add-on/TA-Demisto/default/setup.xml [hunk 5]
```diff
@@ -221,6 +221,7 @@
 		var fieldLabel = "";
 		var ipRegex = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$";
 		var hostRegex = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])*\/{0,1}[A-Za-z0-9][A-Za-z0-9\-]*\_{0,1}[A-Za-z0-9][A-Za-z0-9\-]*$";
+        var urlRegex = "(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/ig";
```
Comment (orius123) [inline add-on/TA-Demisto/default/setup.xml:224]: This is a different regex? We should have exact same validations all over

**Review by orius123 — COMMENTED** [review]
Nice

File: add-on/TA-Demisto/default/app.conf (hunk 4 above)
Comment (hirshbergshachar) [reply add-on/TA-Demisto/default/app.conf:15]: @orius123 no. Don't think we will for this one

File: add-on/TA-Demisto/bin/demisto_setup.py (hunk 1 above)
Comment (hirshbergshachar) [reply add-on/TA-Demisto/bin/demisto_setup.py]: @orius123 Fixed to match the other url regex

**Comment by orius123** [comment]
:+1:

**Pull request merged**